packages/@apphosting/adapter-nuxt/src/utils.ts

```diff
@@ -0,0 +1,128 @@
+import type { NuxtOptions } from "@nuxt/schema";
+import fsExtra from "fs-extra";
+import { spawnSync } from "node:child_process";
+import { writeFile } from "node:fs/promises";
+import { join, normalize, posix, relative } from "node:path";
+import { stringify as yamlStringify } from "yaml";
+
+import { DEFAULT_COMMAND } from "./constants.js";
+import type { OutputBundleOptions } from "./interfaces.js";
+
+let nuxtOptions: NuxtOptions;
+export async function getConfig(cwd: string): Promise<NuxtOptions> {
+  if (!nuxtOptions) {
+    const { loadNuxtConfig } = await import("@nuxt/kit");
+    nuxtOptions = await loadNuxtConfig({ cwd });
+  }
+
+  return nuxtOptions;
+}
+
+async function getWantsBackend(cwd: string): Promise<boolean> {
+  const { ssr: wantsBackend } = await getConfig(cwd);
+
+  return wantsBackend;
+}
+
+export async function build(cwd: string, cmd = DEFAULT_COMMAND): Promise<void> {
+  const command = (await getWantsBackend(cwd)) ? "build" : "generate";
+
+  const build = spawnSync(cmd, ["run", command], {
+    cwd,
+    stdio: "inherit",
+    env: { ...process.env, NITRO_PRESET: "node" },
+  });
+
+  if (build.status !== 0) throw Error("Was unable to build your Nuxt application.");
+}
+
+/**
+ * Moves the server and client code generated by nuxt build into the output directory
+ * Also generates the bundle.yaml file.
+ * @param cwd The current working directory
+ * @param outputBundleOptions The target location of built artifacts in the output bundle.
+ * @param outDir The location of the dist directory.
+ */
+export async function generateOutputDirectory(
+  cwd: string,
+  outputBundleOptions: OutputBundleOptions,
+): Promise<void> {
+  const outDir = join(cwd, ".output");
+
+  await fsExtra.copy(outDir, outputBundleOptions.outputDirectory, {
+    overwrite: true,
+  });
+  // TODO: review, is this needed? Currently having errors with this
+  // await fsExtra.copy(
+  //   join(cwd, "node_modules"),
+  //   join(outputBundleOptions.outputDirectory, "node_modules"),
+  //   { overwrite: true },
+  // );
+
+  await Promise.all([generateBundleYaml(outputBundleOptions, cwd)]);
```

Comment:
Do we need promise.all if this is just a single promise?

Reply:
definitely no, moved the output copy here